pkg/webui/components/portal/index.js

```diff
@@ -0,0 +1,63 @@
+// Copyright © 2024 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { useEffect } from 'react'
+import DOM from 'react-dom'
+
+import PropTypes from '@ttn-lw/lib/prop-types'
+
+// This component create a portal to render children into a different part of the DOM.
+const Portal = ({ children, visible, positionReference }) => {
+  const [buttonPosition, setButtonPosition] = React.useState()
+  useEffect(() => {
+    if (positionReference.current && visible) {
+      const rect = positionReference.current.getBoundingClientRect()
+      setButtonPosition(rect)
+      const elementLeft = rect.left + window.scrollX
+      const elementTop = rect.top + window.scrollY
+      const element = document.getElementById('modal-container')
```

Comment:
You shouldn't modify the modal-container, but only use it as a topmost container. If we modify the position of the modal container, it will cause issues when we're rendering modals.

Reply:
Given the special use cases for these portalled dropdowns, I believe it would be better to add a new container `<div id="dropdown-container" />` This can be styled with absolute positioning to render at top: 0, left: 0 and width: 0, height: 0.